ovirt/resource_ovirt_vm.go

```diff
@@ -277,12 +283,19 @@ func resourceOvirtVMDelete(d *schema.ResourceData, meta interface{}) error {
 				return resource.RetryableError(fmt.Errorf("Stop instance timeout and got an error: %v", err))
 			}
 		}
+		detachOnly := true
+		if d.Get("template") != "Blank" {
```

Comment:
IMHO, getting the `template` attribute from a `vm` is more reliable than from terraform state.
```go
detachOnly := true
if template, ok := vm.Template(); ok {
  if template.MustName() != "Blank" {
    detachOnly = false
  }
}
```